File: 프로그래머스/0/120921. 문자열 밀기/문자열 밀기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string A, string B) {
    int answer = 0;
    
    if(A != B)
    {
        answer = 1;
        for(int i = 0; i < A.size() - 1; i++)
        {
            char last = A[A.size() - 1];
            string tail = A.substr(0, A.size() - 1);
            string result = last + tail;
            A = result;
            
            if(result == B)
                break;
            
            answer++;
        }
    }
    return answer > A.size() - 1 ? -1 : answer;
}

File: 프로그래머스/0/120921. 문자열 밀기/문자열 밀기.cs
using System;

public class Solution {
    public int solution(string A, string B) {
        int answer = 0;
        char[] a = A.ToCharArray();
        char[] b = B.ToCharArray();
        while(answer < a.Length)
        {
            bool check = true;
            
            for(int i = 0; i < a.Length; i++)
            {
                if(a[i] != b[i])
                {
                    check = false;
                    break;
                }
            }
            
            if(check)
                break;
            
            answer++;
            char last = a[a.Length - 1];
            for(int i = a.Length - 1; i >= 0; i--)
            {
                if(i == 0)
                    a[0] = last;
                else
                    a[i] = a[i - 1];
            }
        }
        return answer = answer >= a.Length ? -1 : answer;
    }
}